src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java

```diff
@@ -387,6 +392,62 @@ public FeatureDataSource( final File featureFile, final FeatureCodec<T, ?> codec
     }
 
     /**
+     * Returns the sequence dictionary created from the index or null if that fails.
+     */
+    public SAMSequenceDictionary getSequenceDictionary() {
+        SAMSequenceDictionary dict = null;
+        final Object header = getHeader();
+        if (header instanceof VCFHeader) {
+            dict = ((VCFHeader) header).getSequenceDictionary();
+        }
+        return dict == null || dict.isEmpty() ? createSequenceDictionaryFromContigList() : dict;
+    }
+
+    /**
+     * Load the index from disk, checking for out of date indexes and old versions
+     * @return an Index, or null if we're unable to load
+     */
+    private Index loadIndexFromDisk() {
+        if (! hasIndex) {
+            return null;
+        }
+        final File indexFile = Tribble.indexFile(featureFile);
+        if (! indexFile.canRead()) {
+            return null;
+        }
+        logger.debug("Loading Tribble index from disk for file " + featureFile);
+        final Index index = IndexFactory.loadIndex(indexFile.getAbsolutePath());
+
+        // check if the file is up-to date (filestamp and version check)
+        if (index.isCurrentVersion() && indexFile.lastModified() >= featureFile.lastModified()) {
+            return index;
+        } else if (indexFile.lastModified() < featureFile.lastModified()) {
+            throw new UserException("Index file " + indexFile + " is out of date (index older than input file). Use IndexFeatureFile to make an index.");
+        } else { // we've loaded an old version of the index, we want to remove it <-- currently not used, but may re-enable
+            throw new UserException("Index file " + indexFile + " is out of date (old version). Use IndexFeatureFile to make an index.");
+        }
+    }
+
+    /**
+     * get the sequence dictionary contig list that is always in the index
+     * @return a SAMSequenceDictionary or null if the index cannot be loaded or there are no contigs in the index
```

Comment:
line 1 says "contig list that is always in the index". line 2 says "null if...there are no contigs in the index". Which is it, Mr. Kiezun? :)